src/Fakta/Session.fs

```diff
@@ -12,110 +13,141 @@ open Fakta
+    let urlPath = (sprintf "info/%s" sessionID )
+    let uriBuilder = UriBuilder.ofSession state.config urlPath
+    let result = Async.RunSynchronously (call state (sessionDottedPath "info") id uriBuilder HttpMethod.Put)
+    async {
+      match result with 
+      | Choice1Of2 x -> 
+          let body, (dur:Duration, resp:Response) = x
+          let items = if body = "" then [] else Json.deserialize (Json.parse body)
+          let item = if items.Length = 0 then SessionEntry.empty else items.[0]
+          return Choice1Of2 (item, queryMeta dur resp)
+      | Choice2Of2 err -> return Choice2Of2(err)
+    }
+    
 
 /// List gets all active sessions 
 let list (state : FaktaState) (qo : QueryOptions) : Async<Choice<SessionEntry list * QueryMeta, Error>> =
-  raise (TBD "TODO")
+  getSessionEntries "" "list" state qo 
 
 /// List gets sessions for a node
-let node (s : FaktaState) (node : Node) (qo : QueryOptions) : Async<Choice<SessionEntry list * QueryMeta, Error>> =
-  raise (TBD "TODO")
+let node (state : FaktaState) (node : string) (qo : QueryOptions) : Async<Choice<SessionEntry list * QueryMeta, Error>> =
+  getSessionEntries node "node" state qo
 
 /// Renew renews the TTL on a given session
-let renew (s : FaktaState) (id : Session) (wo : WriteOptions) : Async<Choice<SessionEntry * QueryMeta, Error>> =
-  raise (TBD "TODO")
+let renew (state : FaktaState) (sessionID : string) (wo : WriteOptions) : Async<Choice<SessionEntry * QueryMeta, Error>> =
+    let urlPath = (sprintf "renew/%s" sessionID )
+    let uriBuilder = UriBuilder.ofSession state.config urlPath
+    let result = Async.RunSynchronously (call state (sessionDottedPath "renew") id uriBuilder HttpMethod.Put)
+    async {
+      match result with 
+      | Choice1Of2 x -> 
+          let body, (dur:Duration, resp:Response) = x
+          let items = if body = "" then [] else Json.deserialize (Json.parse body)
+          let item = if items.Length = 0 then SessionEntry.empty else items.[0]
+          return Choice1Of2 (item, queryMeta dur resp)
+      | Choice2Of2 err -> return Choice2Of2(err)
+    }
+
 
 /// RenewPeriodic is used to periodically invoke Session.Renew on a session until
 /// a doneCh is closed. This is meant to be used in a long running goroutine
 /// to ensure a session stays valid. 
-let renewPeriodic (s : FaktaState) (initialTTL : Duration) (id : string) (wo : WriteOptions) : Async<Choice<unit, Error>> =
+let rec renewPeriodic (state : FaktaState) (ttl : Duration) (id : string) (wo : WriteOptions) (doneCh : Duration) =
   //func (s *Session) RenewPeriodic(initialTTL string, id string, q *WriteOptions, doneCh chan struct{}) error
-  raise (TBD "TODO")
+  let result = Async.RunSynchronously( renew state id wo )
+  async {
+    let waitDur = (ttl / (int64)2)
+    let ms = (int)waitDur.Ticks/10000
+    do! Async.Sleep(ms) 
+    match result with 
+    | Choice1Of2 x -> 
+        let entry, _ = x
+        renewPeriodic state entry.ttl id wo |> ignore
```

Comment:
As opposed to `Task<T>`, `Async<T>` are cold. If you ignore it it won't be started.